### DISCUSSION TITLE
Response from earthaccess devs about issues importing .tifs

### DISCUSSION THREAD

**Discussion by NobodyLivesHere** [earthlab-education/earth-analytics-02-python-2024#9] [General]
So I had asked on the earthdata forum about the .tif import error that people were getting and it appears to indeed be an authentication issue

Here was the response 

![image](https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/88915377/f0e23547-ea28-40d8-8466-0e895ab600f0)

And here are the links they provided:

https://github.com/nasa/HLS-Data-Resources
https://github.com/nasa/HLS-Data-Resources/blob/main/python/tutorials/HLS_Tutorial.ipynb

I am currently trying to test this out, though my conda environment has been giving me issues with the gdal module.

**Comment by edchanonline** [comment]
I don't fully understand why my configurations are working yet but I figure I'd share in case it'd help you. I am on MacOS (Sonoma Version 14.2.1 (23C71)). Let us know what you end up doing.

Proof that it works.
![Screenshot 2024-02-19 at 10 52 05 AM](https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/4795392/6db10082-554e-43ca-a204-64c6dd9e56b4)

Files that were created in my home directory:
```
ls -al | grep -E ".netrc|cookies.txt"
-rw-r--r--@   1 ed.chan  staff     79 Feb 17 22:39 .netrc
-rw-r--r--@   1 ed.chan  staff   2191 Feb 18 10:35 cookies.txt
```

I have a `.env` file in my repo (make sure you git ignore this one):
```
EARTHDATA_USERNAME=<your username>
EARTHDATA_PASSWORD=<your password>
GDAL_HTTP_MAX_RETRY=5
GDAL_HTTP_RETRY_DELAY=1
GDAL_HTTP_COOKIEFILE='~/cookies.txt'
GDAL_HTTP_COOKIEJAR='~/cookies.txt'
GDAL_DISABLE_READDIR_ON_OPEN='EMPTY_DIR'
CPL_VSIL_CURL_ALLOWED_EXTENSIONS='TIF'
GDAL_HTTP_UNSAFESSL='YES'
```

For that `.netrc` file, you could create by hand or by script:

Code snippet:
```
from dotenv import load_dotenv


username = os.getenv('EARTHDATA_USERNAME')
password = os.getenv('EARTHDATA_PASSWORD')

# earthaccess.login doesn't seem to be creating this file
netrc_fpath = os.path.expanduser("~/.netrc")
if not netrc_fpath:
    with open(netrc_fpath, 'w') as file:
        file.write(f"machine urs.earthdata.nasa.gov\nlogin {username}\npassword {password}\n")
```

Alternatively, you can just manually create that file:
```
machine urs.earthdata.nasa.gov
login <your_username>
password <your_password>
```

**Reply by edchanonline** [reply]
Forgot to say. You had to load the `.env` file too. Here's how:

```
from dotenv import load_dotenv

load_dotenv() # Load .env file
```

**Comment by eculler** [comment]
Nice work @eggvoice @NobodyLivesHere!

I wonder if it works for me because I have downloaded Earthdata files from a command line script in the past, and so probably still have the cookies set up?

@eggvoice if you log in using earthaccess.login it will do the .netrc file for you.

You can also use the os package to set environment variables in the code (although don’t out your password or username in plain text)

**Reply by edchanonline** [reply]
Thanks Elsa! 

I moved all my environment variables to `.env` because I thought I needed to configure `EARTHDATA_PASSWORD`. 

I was using `strategy='environment'` having read that

```
"environment": retrieve username and password from $EARTHDATA_USERNAME and $EARTHDATA_PASSWORD.
        persist (Boolean): will persist credentials in a .netrc file
```

It didn't create that `.netrc` file for whatever reasons so I created that via my snippet. Didn't really have a chance to further troubleshoot that one as I have moved on to the rest of the assignment.

**Reply by edchanonline** [reply]
Ye, it wasn't super clear to me what step in the process the `cookies.txt` was created. I also did some downloading via `earthaccess.download()` and perhaps that's when it was created.
